onnxruntime/core/providers/js/operators/reduce.h

```diff
@@ -29,7 +29,7 @@ namespace js {
                                  static_cast<int32_t>(keepdims_),                                            \
                                  static_cast<int32_t>(noop_with_empty_axes_),                                \
                                  gsl::narrow_cast<int32_t>(axes.size()),                                     \
-                                 reinterpret_cast<int32_t>((axes.size() > 0) ? axes.data() : nullptr) >> 2); \
+                                 JSEP_HEAP_PTR((axes.size() > 0) ? axes.data() : nullptr) >> 2);             \
```

Comment:
Why shift to right by 2 bits? Before the shifting the variable is a pointer.

Reply:
I saw the comment below.

Reply:
if we divide by 4 we use wasm.HEAP32[] on the javascript side